misc/python/materialize/parallel_workload/parallel_workload.py

```diff
@@ -83,20 +83,21 @@ def run(
     conn = pg8000.connect(host=host, port=port, user="materialize")
     conn.autocommit = True
     with conn.cursor() as cur:
-        database.create(Executor(rng, cur))
+        database.create(Executor(rng, conn, cur))
     conn.close()
 
     conn = pg8000.connect(
         host=host, port=port, user="materialize", database=str(database)
     )
     conn.autocommit = True
     with conn.cursor() as cur:
-        database.create_relations(Executor(rng, cur))
+        database.create_relations(Executor(rng, conn, cur))
     conn.close()
 
     workers = []
     threads = []
     for i in range(num_threads):
+        # TODO: check with Philip: move outside loop?
```

Comment:
@philip-stoev

Reply:
I believe that they still get random numbers because this is call-by-reference. That is, each thread won't get a copy of this random but rather operate on the same random instance.

This example shows that threads won't produce the same numbers:
```
import random
import threading

worker_rng = random.Random(3)

class MyT:
  def __init__(self, t: str, worker_rng: random.Random):
    self.t = t
    self.worker_rng = worker_rng
  def run(self) -> None:
    for i in range(1, 10):
      print(f"{self.t} produces: {self.worker_rng.randint(0, 10)}")

for t in ["A", "B", "C"]:
  x = MyT(t, worker_rng)
  thread = threading.Thread(target=x.run)
  thread.start()
```